src/jvm/main/org/jetbrains/kotlinx/lincheck/IdeaPlugin.kt

```diff
@@ -79,7 +79,8 @@ fun shouldReplayInterleaving(): Boolean {
  */
 @Suppress("UNUSED_PARAMETER")
 fun beforeEvent(eventId: Int, type: String) {
-    val strategy = (Thread.currentThread() as? TestThread)?.eventTracker ?: return
+    val strategy = Injections.getCurrentThreadDescriptor()?.eventTracker
```

Comment:
Have you checked how custom threads work with the plugin?

Reply:
No it does not currently work. 

I found one problem in the plugin code that I've managed to fix by myself, but it still does not work. 
There are some other bugs, and I need help from someone on the plugin side to debug the remaining problems.

Reply:
The plugin seems to work now, but it required a few commits to the plugin repo. We would need to sync the releases of the new version of lincheck and the plugin.